server/src/main/java/org/apache/iotdb/db/query/udf/service/UDFRegistrationService.java

```diff
@@ -216,7 +216,7 @@ public void deregister(String functionName) throws UDFRegistrationException {
       logger.error(errorMessage);
       throw new UDFRegistrationException(errorMessage);
     }
-
+    registrationInformation.remove(functionName);
```

Comment:
I think it's better to move the `remove` statement to the last. What do you think?  :D

```suggestion
    if (!information.isTemporary()) {
      try {
        appendDeregistrationLog(functionName);
      } catch (IOException e) {
        registrationInformation.put(functionName, information);
        String errorMessage =
            String.format(
                "Failed to append UDF log when deregistering UDF %s, because %s", functionName, e);
        logger.error(errorMessage);
        throw new UDFRegistrationException(errorMessage, e);
      }
    }
    
    registrationInformation.remove(functionName);
```

Reply:
@SteveYurongSu  good,  code should be the same as master branch.  it's better to move the remove statement to the last.

Reply:
@SteveYurongSu That's the best solution! A new commit is pushed.